### DISCUSSION TITLE
Problemi con installazione HACS su HA Core e Raspberry

### DISCUSSION THREAD

**Discussion by Manoel81** [indomus/forum#2613] [HUB: Home Assistant]
Buongiorno a tutti.

Piccola premessa: sono uno smanettone al quale piace lanciarsi in progetti ambiziosi, a volte anche troppo complicati. Sono, in particolare, nuovissimo del mondo HA, nonché privo di chissà quali nozioni del mondo Linux.

Volendo provare HA e avendo in casa un Raspberry a disposizione, ho deciso di provare questo HUB e ieri ho installato su una SD Raspbian con Docker (e Docker Compose) e HA Core, seguendo le mille guide di Indomus. Sembra che tutto sia andato per il verso giusto, perché riesco ad accedere al sistema regolarmente.

Perché dico "sembra"? Perché quando ho provato ad installare HACS tramite script usando il terminale, ho ricevuto il seguente errore "could not find the directory for Home Assistant". Sbaglio qualcosa?
Provando la procedura manuale, sempre contenuta nella stessa guida di indomus, viene detto di creare una cartella "hack" all'interno della cartella "custom_components", che però non trovo. Questa cartella, a quanto ho capito, dovrebbe contenere anche il file YAML di configurazione di HA, quindi da qualche parte dovrebbe esserci, visto che come detto HA gira correttamente.

Perdonatemi se il problema vi sembra banale, ma essere alle prime armi anche in ambiente Linux, per me ogni passaggio risulta complesso e magari mi sto perdendo in un bicchiere d'acqua.

Grazie a tutti per ogni tipo di aiuto.

**Comment by indomus** [comment]
> quando ho provato ad installare HACS tramite script usando il terminale, ho ricevuto il seguente errore "could not find the directory for Home Assistant". Sbaglio qualcosa? 

in che punto[ della guida](https://indomus.it/guide/installare-hacs-home-assistant-community-store-sul-proprio-hub/)?

**Reply by snakuzzo** [reply]
Se hai ricevuto quell'errore, vuol dire che non sei nella directory corretta.

Se sei sull'host, prima devi posizionarti nella directory dove hai la configurazione di Home Assistant e poi scarichi ed esegui lo script. 
Oppure entri nel container con `docker exec -it <nomecontainerhomeassistant> bash`, ti porti sotto `/config` e da qui scarichi ed esegui lo script

```
wget -O - https://get.hacs.xyz | bash -
```

La procedura manuale prevede che:

- Crei la directory `hacs` (e non `hack` come hai scritto tu...anche se alla fine non ha tanta importanza) sotto la config dir di home assistant 
- Ti posizioni sotto questa directory
- Scarichi il file zip da github ( https://github.com/hacs/integration/releases/latest ) e poi lo scompatti sotto la directory `hacs` appena creata 
- Riavvii Core
 
Fine

**Comment by Manoel81** [comment]
> in che punto[ della guida](https://indomus.it/guide/installare-hacs-home-assistant-community-store-sul-proprio-hub/)?

Nel momento in cui vado ad installare HA Core su Docker coi comandi:

docker exec -it home-assistant bash
wget -O - https://get.hacs.xyz | bash -
exit

Però io l'ho lanciato dalla root principale, non sono entrato in nessuna sottocartella.

**Reply by snakuzzo** [reply]
Leggi cosa ti ho scritto

**Reply by snakuzzo** [reply]
E, per favore, non creare più thread nella stessa discussione (come hai appena fatto) :)
Fai replay sotto l'ultima risposta.
Grazie

**Reply by Manoel81** [reply]
Ho solo quotato il post al quale stavo rispondendo. Non va bene?

**Comment by Manoel81** [comment]
> Se hai ricevuto quell'errore, vuol dire che non sei nella directory corretta.
> 
> Se sei sull'host, prima devi posizionarti nella directory dove hai la configurazione di Home Assistant e poi scarichi ed esegui lo script. Oppure entri nel container con `docker exec -it <nomecontainerhomeassistant> bash`, ti porti sotto `/config` e da qui scarichi ed esegui lo script
> 
> ```
> wget -O - https://get.hacs.xyz | bash -
> ```

Io ho creato una cartella che si chiama "homeassistant" come da guida ma, lanciando il comando ls, questa sembra vuota. E' normale?

> La procedura manuale prevede che:
> 
> * Crei la directory `hacs` (e non `hack` come hai scritto tu...anche se alla fine non ha tanta importanza) sotto la config dir di home assistant
> * Ti posizioni sotto questa directory
> * Scarichi il file zip da github ( https://github.com/hacs/integration/releases/latest ) e poi lo scompatti sotto la directory `hacs` appena creata
> * Riavvii Core
> 
> Fine

Appena arrivo a casa riprovo con lo script, se non dovesse andare tenterò la procedura manuale.

**Reply by snakuzzo** [reply]
stai facendo parecchia confusione.
Ti ho detto cosa devi creare per hacs.
Tu mi stai parlando della dir di homeassistant dove, ammesso tu abbia correttamente configurato il bind dei volumi quando hai creato il container, dovresti avere tutti i file di configurazione di Core

**Reply by Manoel81** [reply]
Aspetta un attimo. Io ho lanciato lo script per HACS dalla root principale e tu mi hai detto di farlo invece dalla cartella che contiene la configurazione di HA, che credevo fosse appunto quella denominata "homeassistant". Non è così?

**Reply by snakuzzo** [reply]
Devi lanciarlo dalla directory in cui hai i file di configurazione di Home Assistant. Io non posso sapere se fin qui hai fatto il bind dei volumi correttamente oppure no e come hai chiamato le directory

**Reply by Manoel81** [reply]
Io mi sono limitato a seguire il più fedelmente possibile la guida,  proprio per non incappare in errori. Però capisco benissimo la tua difficoltà nell'aiutarmi.

Davo per scontato che il fatto che HA si avviasse correttamente fosse indice di una corretta installazione.

In sostanza, ho installato Docker e Docker Compose correttamente, ho modificato il file Docker-compose.yaml con la configurazione di HA, ho istanziato il container e basta. Così HA funziona, volendo poi aggiungere HACS sono incappato nell'errore in questione.

**Reply by snakuzzo** [reply]
Se posso permettermi di darti un consiglio spassionato... cerca di studiare un attimino come funziona docker e soprattutto come funziona il bind dei volumi, perché se non capisci bene questo concetto, non saprai mai se stai facendo la cosa giusta oppure no.
Se dici di essere uno "smanettone", come leggo nel primo messaggio, di sicuro non avrai difficoltà.

**Reply by Manoel81** [reply]
Io sinceramente sto perdendo le speranze. Ho passato giorni a leggere guide, ma relativamente a questo problema ne so come prima.

Primo problema: in questo sito ci sono due guide diverse che specificano come editare il file Docker-compose.yaml per una corretta installazione di HomeAssistant.
La prima guida è questa: https://indomus.it/formazione/gestire-applicazioni-docker-tramite-docker-compose/
mentre la seconda è questa: https://indomus.it/guide/come-installare-e-configurare-home-assistant-core-con-docker-su-un-raspberry-pi-gia-in-uso/

I due file sono però diversi. Quale dei due devo utilizzare?

**Reply by snakuzzo** [reply]
Quelle due guide sono due cose completamente diverse...eppure mi sembra abbastanza chiaro.
Nella prima viene descritto in generale come gestire applicazioni con docker-compose. La seconda è invece una guida per l'installazione su un RPI. Anche il titolo mi sembra sia abbastanza parlante.

Quanto al consiglio che ti ho dato nel messaggio dell'altro giorno, ovvero di studiare come funziona docker, non è detto che tu debba cercare sul nostro sito i manuali per studiare docker. Gli articoli che stai menzionando non sono dei manuali né documentazione ufficiale con cui studiare. Se vuoi apprendere li funzionamento di docker, vai sul sito ufficiale e studia. Ci sono anche canali youtube dove spiegano come funziona. Fai prove, smanetta (hai detto di saperlo fare) e chiarisci bene soprattutto quello che ti ho già indicato. Ti dico questo perché noi non siamo qui ad insegnare come funziona docker, spero che questo sia chiaro.
Non puoi pretendere di non sapere nulla di docker, seguire una guida e poi avere già imparato tutto.

**EDIT:** P.S. L'installazione di HACS in tutto ciò non c'entra niente

**Reply by Manoel81** [reply]
Non mi trattare con sufficienza, perché il fatto che io non sia esperto della materia qui trattata non indica che io sia uno stupido. Potrò magari perdermi in aspetti che tu reputi sciocchi, ma di sicuro se sono su un forum a cercare aiuto è perché sto cercando di imparare.

Io non ho mai detto di aver letto solo guide di questo sito. Non l'ho fatto, anche perché, come dici giustamente tu, la materia trattata qui è un'altra. Infatti io ho chiesto qui per un errore nell'installazione di HACS su HA. Direi che sono sul sito giusto, no?
Tu mi hai risposto che il problema sta in Docker, e posso anche crederci, ma avendo io seguito fedelmente le guide qui presenti, mi sono permesso di chiedere lumi. Guide che non partono da HA, ma partono addirittura dalla scelta dell'hardware e dall'installazione del sistema operativo. Quindi se ho problemi con una vostra guida, dove altro avrei dovuto chiedere aiuto?

E non ho mai detto di aver imparato tutto, non sarebbe lontanamente vero, anzi sono stato il primo ad ammettere le mie lacune.

PS: l'errore durante l'installazione di HACS è stato il primo problema finora riscontrato. Se anche il problema non fosse in HACS, ma in quello che sta a monte, direi che le due cose sono legate. E in ogni caso, che il problema stia in Docker me l'hai detto tu, e per questo ti ringrazio, ma è sintomo che evidentemente un legame tra le due cose c'è eccome. Ad iniziare proprio dalla guida.

**Reply by indomus** [reply]
Perdonami, non ti sta trattando con sufficienza. Ti sta garbatamente facendo presente che stai confondendo le cose. Devi riuscire a chiarirti:
- cosa sia un container docker
- come si istanzi con docker run oppure con docker compose
- eventualmente come si gestiscano i vari contesti.

Questo è: in larga parte è spiegato sul sito, ma molto altro lo trovi in rete. Se diventa troppo complicato per te, forse la strada giusta è Home Assistant OS, il quale ti solleva tali complessità e rende tutto molto più fruibile e semplice.

**Comment by fradf75** [comment]
Buonasera, sono completamente asciutto ed ho un problema con l'installazione di HACS.
Qualcuno potrebbe aiutarmi?

**Reply by snakuzzo** [reply]
se non scrivi che problema hai uno come fa ad aiutarti ?

**Reply by fradf75** [reply]
scusa, ho installato hacs, ma non apre nessuna finestra

**Reply by fradf75** [reply]
ho cercato un pò e su un altro forum mi hanno fatto cancellare la custom qualcosa
adesso non riesco più a trovarlo

**Reply by snakuzzo** [reply]
Cosa si dovrebbe aprire (fammi capire) ?
Inoltre, cosa vuol dire che ti hanno detto di cancellare la "custom qualcosa" ? Parli della directory `custom_components` ? In tal caso, se hai già altra roba rischi di cancellare pure il resto...e comunque è una "NON soluzione" perché ti limiteresti semplicemente a cancellare il custom component.

Io capisco che magari sei alle prime armi, ma forse sei troppo alle prime armi. L'installazione di HACS è una delle cose più semplici (anche più semplice dell'installazione di Home Assistant)

**Reply by fradf75** [reply]
Adesso ho installato nuovamente, ma compare sempre la stessa cosa.
allego foto
![Uploading Screenshot 2023-12-14 alle 18.54.34.png…]()

**Reply by fradf75** [reply]
più che alle prime armi, direi crudo

**Comment by riky78** [comment]
io ho il problema che hacs non mi aggiunge l'integrazione... non so se sbaglio qualcosa o ho un problema con hacs.
Nello specifico trovo l'integrazione su HACS la scarico poi vado su impostazioni, dispositivi e servizi,+ aggiungi integrazione ma non lo trova... ho riavviato HA, cancellato chache ma nulla... grazie a chi può essere d'aiuto
